Block/NodeType/CustomUrl.php

```diff
@@ -81,14 +81,31 @@ public function getHtml($nodeId, $level)
     {
         $classes = $level == 0 ? 'level-top' : '';
         $node = $this->nodes[$nodeId];
-        $url = $this->_storeManager->getStore()->getBaseUrl() . $node->getContent();
+        $nodeContent  = $node->getContent();
         $title = $node->getTitle();
+        $target = $this->getTarget() == 1 ? 'target="_blank"' : '';
+
+        if (!$this->isExternalUrl($nodeContent)) {
+            $url = $this->_storeManager->getStore()->getBaseUrl() . $nodeContent;
+        } else {
+            $url = $nodeContent;
+        }
 
         return <<<HTML
-<a href="$url" class="$classes" role="menuitem"><span>$title</span></a>
+<a href="$url" class="$classes" $target role="menuitem"><span>$title</span></a>
 HTML;
     }
 
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    private function isExternalUrl(string $url = '')
```

Comment:
The default value `$url` is not needed.  BTW are you sure the `$node->getContent()` method always return `string`? I ask because you use type hint and it will cause fatal error for `null` as the parameter.

Reply:
Done